src/main/java/hudson/plugins/claim/ClaimConfig.java

```diff
@@ -53,55 +67,66 @@ public boolean configure(StaplerRequest req, JSONObject formData) throws FormExc
     public boolean getSendEmails() {
         return sendEmails;
     }
-    
+
     /**
      * Set whether we should send emails
      * @param val the setting to use
      */
     public void setSendEmails(boolean val) {
         sendEmails = val;
     }
-    
+
     /**
      * Returns true if the claims should be sticky by default, false otherwise.
-     * 
+     *
      * @return true to make claims sticky by default, else false.
      */
     public boolean isStickyByDefault() {
-	return stickyByDefault;
+        return stickyByDefault;
     }
 
     /**
      * Sets the default stickiness behaviour for build claims.
-     * 
+     *
      * @param stickyByDefault
      *            the default stickiness value.
      */
     public void setStickyByDefault(boolean stickyByDefault) {
-	this.stickyByDefault = stickyByDefault;
+        this.stickyByDefault = stickyByDefault;
     }
 
-    /**
-     * This method returns the Groovy script as a String
-     * @return String containing the Groovy script to run when claims are changed.
-     */
-    public String getGroovyScript() {
-        return groovyScript;
+    @CheckForNull
+    public SecureGroovyScript getGroovyTrigger() {
+        return groovyTrigger;
     }
-    
-    /**
-     * Set the Groovy script to run when a claim is changed.
-     * @param val the script to use
-     */
-    public void setGroovyScript(String val) {
-        groovyScript = val;
+
+    void setGroovyTrigger(SecureGroovyScript groovyTrigger) {
+        if (groovyTrigger == null) {
+            this.groovyTrigger = null;
+        } else {
+            this.groovyTrigger = groovyTrigger.configuringWithKeyItem();
```

Comment:
`configuringWithKeyItem` seems to make no sense here. This is a `GlobalConfiguration`, so there will be no associated `Item`.

I think you just meant to use `.configuring(ApprovalContext.create().withCurrentUser())`. You could I guess add `withKey(SOME_CONSTANT)` if you care about canceling previous request.